jupyter/d2l-java/chapter_multilayer-perceptrons/dropout.ipynb

```diff
@@ -0,0 +1,812 @@
+   "metadata": {
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "2"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "NDArray X = manager.arange(16).reshape(2, 8);\n",
+    "System.out.println(dropoutLayer(X, 0));\n",
+    "System.out.println(dropoutLayer(X, 0.5f));\n",
+    "System.out.println(dropoutLayer(X, 1.0f));"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Defining Model Parameters\n",
+    "\n",
+    "Again, we work with the Fashion-MNIST dataset\n",
+    "introduced in :numref:`sec_softmax_scratch`.\n",
+    "We define a multilayer perceptron with \n",
+    "two hidden layers containing 256 outputs each."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "3"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "int numInputs = 784;\n",
+    "int numOutputs = 10;\n",
+    "int numHiddens1 = 256;\n",
+    "int numHiddens2 = 256;\n",
+    "\n",
+    "NDArray W1 = manager.randomNormal(0, 0.01f, new Shape(numInputs, numHiddens1), DataType.FLOAT32, Device.defaultDevice());\n",
+    "NDArray b1 = manager.zeros(new Shape(numHiddens1));\n",
+    "NDArray W2 = manager.randomNormal(0, 0.01f, new Shape(numHiddens1, numHiddens2), DataType.FLOAT32, Device.defaultDevice());\n",
+    "NDArray b2 = manager.zeros(new Shape(numHiddens2));\n",
+    "NDArray W3 = manager.randomNormal(0, 0.01f, new Shape(numHiddens2, numOutputs), DataType.FLOAT32, Device.defaultDevice());\n",
+    "NDArray b3 = manager.zeros(new Shape(numOutputs));\n",
+    "\n",
+    "\n",
+    "NDList params = new NDList(W1, b1, W2, b2, W3, b3);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Defining the Model\n",
+    "\n",
+    "The model below applies dropout to the output \n",
+    "of each hidden layer (following the activation function).\n",
+    "We can set dropout probabilities for each layer separately.\n",
+    "A common trend is to set\n",
+    "a lower dropout probability closer to the input layer.\n",
+    "Below we set it to 0.2 and 0.5 for the first \n",
+    "and second hidden layer respectively.\n",
+    "By using the `isTraining` boolean variable described in :numref:`sec_autograd`,\n",
+    "we can ensure that dropout is only active during training."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "attributes": {
+     "classes": [],
+     "id": "",
+     "n": "4"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "float dropout1 = 0.2f;\n",
+    "float dropout2 = 0.5f;\n",
+    "\n",
+    "public NDArray net(NDArray X, boolean isTraining) {\n",
+    "\n",
+    "    X = X.reshape(-1, numInputs);\n",
+    "    NDArray H1 = Activation.relu(X.dot(W1).add(b1));\n",
+    "\n",
+    "    if (isTraining) {\n",
+    "        Dropout dp1 = Dropout.builder().optProbability(dropout1).build();\n",
```

Comment:
You are not using the DropoutLayer function to train

Reply:
I'm actually using it only to train. If you look into the epoch loop, I've passed true when training and false while validating.

Reply:
> I'm actually using it only to train. If you look into the epoch loop, I've passed true when training and false while validating.

I mean, the `DropoutLayer` function you defined above. This Dropout is the DJL implementation